libkbfs/block_journal.go

```diff
@@ -746,6 +781,24 @@ func (j *blockJournal) removeFlushedEntry(ctx context.Context,
 		return errBlockJournalShutdown
 	}
 
+	// Fix up the block byte count if we've finished a Put.
+	if entry.Op == blockPutOp {
+		id, _, err := entry.getSingleContext()
+		if err != nil {
+			return err
+		}
+		b, err := j.getDataSize(id)
+		if err != nil {
+			return err
+		}
+
+		if b > j.blockBytes {
+			return fmt.Errorf("Block %v is bigger than our current count "+
+				"of journal block bytes (%d > %d)", id, b, j.blockBytes)
+		}
+		j.blockBytes -= b
```

Comment:
Should be fine as is, I think; this function is called under both `flushLock` and `journalLock`.